torchqdynamics/mesolve.py

```diff
@@ -81,20 +121,31 @@ def forward_adjoint(self, t: float, dt: float, phi: torch.Tensor):
 class MERouchon1_5(MERouchon):
     def forward(self, t: float, dt: float, rho: torch.Tensor):
         """Compute rho(t+dt) using a Rouchon method of order 1.5."""
+        # Args:
+        #     rho: (b_H, b_rho0, n, n)
+        #
+        # Returns:
+        #     (b_H, b_rho0, n, n)
+
         # non-hermitian Hamiltonian at time t
-        H_nh = self.H(t) - 0.5j * self.sum_nojump
+        H_nh = self.H - 0.5j * self.sum_nojump  # (b_H, 1, n, n)
 
         # build time-dependent Kraus operators
-        M0 = self.I - 1j * dt * H_nh
-        Ms = torch.cat((M0[None, ...], sqrt(dt) * self.jump_ops))
+        M0 = self.I - 1j * dt * H_nh  # (b_H, 1, n, n)
+        Ms = sqrt(dt) * self.jump_ops  # (1, len(jump_ops), n, n)
 
         # build normalization matrix
-        S = M0.adjoint() @ M0 + dt * self.sum_nojump
-        S_inv_sqrtm = inv_sqrtm(S)
+        S = M0.adjoint() @ M0 + dt * self.sum_nojump  # (b_H, 1, n, n)
+        # TODO: fix `inv_sqrtm` (size not compatible and linalg.solve RuntimeError)
+        S_inv_sqrtm = inv_sqrtm(S)  # (b_H, 1, n, n)
```

Comment:
The `inv_sqrtm` function was bugged, I propose to fix this in a later PR. Then we also need to make it so it handles properly the new tensor shape.
```python
RuntimeError: linalg.solve: Expected A and B to have the same dtype, but found A of type ComplexDouble and B of type Double instead
```

Reply:
I will soon add tests to `mesolve` because it feels necessary.

Reply:
Yes sorry I have to admit I coded without checking too much. I was like "it's fine we'll make some test function at some point and debug everything". I don't know if that's a good way of working or not haha.